src/main/java/Duke.java

```diff
@@ -1,10 +1,414 @@
+import java.util.Date;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private static Storage storage;
+    private TaskList inputs;
+    private UI ui;
+
+    /*
+     * Constructs Duke Object, constructs UI, storage and TaskList object to initialize
+     *
+     * @param filepath to store data from the input list
+     */
+    public Duke(String filepath) {
+        ui = new UI();
+        storage = new Storage(filepath);
+        try {
+            inputs = new TaskList(storage.readFile());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            inputs = new TaskList();
+        }
+    }
+
+    /*
+     * Runs the programme, and ends process when user input is "bye"
+     *
+     * @throws DukeException if an exception arises while reading input or creating parser object
+     */
+    public void run() throws DukeException {
+        while (true) {
+            String nextLine = ui.readInput();
+            Parser parser = new Parser(nextLine, inputs);
+            if (nextLine.equals("bye")) {
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            }
+            parser.parse();
+        }
+    }
+
+    public static void main (String[]args) throws DukeException {
+        new Duke("listStore.ser").run();
+    }
+
+    // Input class represents user inputted tasks
+    public static class Input implements Serializable {
+        boolean done;
+        String content;
+        String id;
+        LocalDate time;
+        String printTime;
```

Comment:
Is it better to understand a variable or method if you use noun for variable/class and verb for methods/functions?
🤔